report_docx/report/report_docx.py

```diff
@@ -0,0 +1,316 @@
+# -*- coding: utf-8 -*-
+# © 2018 Elico Corp (https://www.elico-corp.com).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo.report import report_sxw
+from odoo import api
+import logging
+from pyPdf import PdfFileWriter, PdfFileReader
+from reportlab.pdfgen import canvas
+import os
+import base64
+from lxml import etree
+import time
+import random
+
+try:
+    from docxtpl import DocxTemplate
+except ImportError:
+    pass
+try:
+    from docx import Document
+except ImportError:
+    pass
+
+_logger = logging.getLogger(__name__)
+
+
+class ReportDocx(report_sxw.rml_parse):
+
+    @api.model
+    def create(self, data):
+        report_id_list = self.env['ir.actions.report.xml'].search([
+            ('report_name', '=', self.name[7:])])
+        if report_id_list:
+            report_xml = report_id_list[0]
+            self.title = report_xml.name
+            if report_xml.report_type == 'docx':
+                return self.create_source_docx(data)
+        return super(ReportDocx, self).create(data)
+
+    @api.multi
+    def create_source_docx(self, dict):
+        data = self.generate_docx_data(dict)
```

Comment:
if using api.multi, you need Verifies that the current recorset holds a single record or make a loop.  
Please refer to other examples.